src/interfaces/swig/SGBase.i

```diff
@@ -485,6 +491,29 @@ namespace shogun
         }
 
         /*int getbuffer(PyObject *obj, Py_buffer *view, int flags) { return 0; }*/
+
+        std::vector<std::string> parameter_names() const {
+			std::vector<std::string> result;
+			for (auto const& each: self->get_parameters()) {
+				result.emplace_back(each.first);
+			}
+			return result;
+        }
+
+		PyObject* parameter_types() const {
+			PyObject* py_result = PyDict_New();
+			for (auto const& each: self->get_parameters()) {
+				int result = PyDict_SetItem(py_result,
+											PyUnicode_FromString(each.first.c_str()),
```

Comment:
Indentation looks strange over there

Reply:
oh wow, it does seem strange..

Reply:
I also need to to add PyString_FromString for python 2